src/CoreWCF.Kafka.Client/src/CoreWCF/ServiceModel/Channels/KafkaOutputChannel.cs

```diff
@@ -136,6 +138,22 @@ private async Task SendAsync(Message message, TimeSpan timeout)
             }
         }
 
+        private static void ApplyKafkaMessageProperty(Message message, Message<byte[], byte[]> kafkaMessage)
+        {
+            KafkaMessageProperty kafkaMessageProperty =
+                message.Properties.TryGetValue(KafkaMessageProperty.Name, out object value) &&
+                value is KafkaMessageProperty property
+                    ? property
+                    : new KafkaMessageProperty();
+            kafkaMessage.Headers = new();
+            foreach (KafkaMessageHeader kafkaMessageHeader in kafkaMessageProperty.Headers)
+            {
+                kafkaMessage.Headers.Add(kafkaMessageHeader.Key, kafkaMessageHeader.Value);
+            }
+
+            kafkaMessage.Key = kafkaMessageProperty.PartitionKey;
+        }
```

Comment:
Is setting a null value here okay?

Reply:
Yes the existing unit tests cover that case.
Please see [here](https://github.com/confluentinc/confluent-kafka-dotnet/blob/a67bd6c06b7eef4293e6476d9ff6f3e93f0e4cd9/src/Confluent.Kafka/Serializers.cs#L170) for serialization side and [here](https://github.com/confluentinc/confluent-kafka-dotnet/blob/a67bd6c06b7eef4293e6476d9ff6f3e93f0e4cd9/src/Confluent.Kafka/Deserializers.cs#L243) for deserialization side.